internal/lock.go

```diff
@@ -150,28 +242,32 @@ func (l *Lock) Download(dir string, tags []string, notags []string, perm string,
 		return fmt.Errorf("nothing to download")
 	}
 	errorCh := make(chan error, total)
-
-	var statusLine *StatusLine
-	if status {
-		statusLine = NewStatusLine(ctx, &filteredResources)
-		err := statusLine.InitResourcesSizes()
-		if err == nil {
-			statusLine.Start(true)
-		} else {
-			statusLine = nil // Do not update or display SL.
-		}
-	}
-
-	for i, r := range filteredResources {
+	for _, r := range filteredResources {
 		resource := r
 		go func() {
+			// See if the resource has an Artifactory URL
+			if resource.CacheUri != "" {
+				// Find the correct filename
+				filename := resource.Filename
+				if filename == "" {
+					filename = path.Base(resource.Urls[0])
+				}
+				// Build the full file path
+				fullPath := filepath.Join(dir, filename)
+
+				err := downloadFromArtifactory(ctx, resource.CacheUri, resource.Integrity, fullPath, mode)
+				if err == nil {
+					errorCh <- nil
+					return
+				}
+				// Show a warning only for connection errors
+				if strings.Contains(err.Error(), "lookup invalid") || strings.Contains(err.Error(), "dial tcp") {
+					fmt.Printf("Failed to download from Artifactory, falling back to original URL: %v\n", err)
+				}
+			}
 
 			err := resource.Download(dir, mode, ctx)
 			errorCh <- err
-
-			if statusLine != nil {
-				statusLine.Increment(i)
-			}
```

Comment:
Same question as above: if this was just added in #24, you shouldn't be removing it.